daemon/monitor.go

```diff
@@ -39,7 +39,7 @@ func (daemon *Daemon) handleContainerExit(c *container.Container, e *libcontaine
 
 	tsk, ok := c.Task()
 	if ok {
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
Shouldn't this context cancellation be separate from the `ctx` passed to this function?
Otherwise, if `handleContainerExit` gets cancelled, the context passed to `tsk.Delete` will already be "done" and the task won't be deleted.